theano/tensor/basic.py

```diff
@@ -4268,15 +4268,51 @@ def tile(x, reps, ndim=None):
     TODO: expand this.
     """
 
+    # if reps is tensor.scalar, integer or tensor.vector, we convert it to a list.
+    if not isinstance(reps, (list, tuple)):
+        reps_astensor = as_tensor_variable(reps)
+        ndim_check = reps_astensor.ndim
+
+        # tensor.scalar/integer case
+        if ndim_check == 0:
+            reps_ = []
+            reps_.append(reps)
+            reps = reps_
+        # tesor.vector case
+        elif ndim_check == 1:
+            if ndim == None:
+                raise ValueError("if reps is tensor.vector, you should specify "
+                                 "the ndim")
+            else:
```

Comment:
Add a comment explaining what we are trying to do here (dynamically add 1 to the left if `reps` is too short)